graphbolt/include/graphbolt/fused_sampled_subgraph.h

```diff
@@ -114,6 +116,12 @@ struct FusedSampledSubgraph : torch::CustomClassHolder {
    * subgraph.
    */
   torch::optional<torch::Tensor> type_per_edge;
+
+  /**
+   * @brief Offsets of each etype,
+   * type_per_edge[etype_offsets[i]: etype_offsets[i + 1]] == i
+   */
+  torch::optional<torch::Tensor> etype_offsets;
```

Comment:
I am a little bit confused here, is type_per_edge guaranteed to be sorted? Is it possible that type_per_edge is actually in a random order?

Reply:
@peizhou001 can you double check the code here, esp. when we implement temporal, some of the assumption we made may break.

Reply:
After I thought about it, maybe the documentation is not accurate. Does etype_offsets' length equal to #etype?

Reply:
When we change the CPU implementation to be aligned with the CUDA implementation, it will be guaranteed to be sorted. That is why we are able to simply slice the final result to convert to sampled subgraphs.

Reply:
etype_offsets' length equal to (#etype + 1)

Reply:
indptr length is equal to 1 + \sum_{etype} #seeds with dst_node_type(etype)

Reply:
It is no longer equal to 1 + #seeds.

Reply:
When we use offsets, the type_per_edge output is no longer required, so we can remove it when we implement the same logic for the CPU implementation.

Reply:
I see, can you add a comment there, this is getting complicated while there are 2 different indptr and requires different code to handle.

Reply:
Added.